tests/verilog/sign_array_query.ys

```diff
@@ -0,0 +1,54 @@
+logger -expect-no-warnings
+
+read_verilog -formal <<EOT
+module top(input clk);
+
+    reg [-1:-1] x;
+    reg y = 0;
+
+    always @(posedge clk) begin
+        case ($left(x)) 36'shfffffffff: y = 1; endcase
+        assert (y);
+    end
+
+endmodule
+EOT
+
+prep -top top
+sim -n 3 -clock clk
+
+design -reset
+read_verilog -formal <<EOT
+module top(input clk);
+
+    reg [-1:-1] x;
+    reg y = 0;
+
+    always @(posedge clk) begin
+        case ($left(x)) 36'sh0ffffffff: y = 1; 32'h0: ; endcase
```

Comment:
It seems like this should pass regardless of whether or not `$left(x)` is signed.

Reply:
That's true, assuming the signedness rules are properly implemented. This test fails when you apply the fix to genrtlil.cc but just change `newNode = mkconst_int(result, false);` to `newNode = mkconst_int(result, true);` instead of `newNode = mkconst_int(result, backup_sign_hint);` in simplify.cc. I wanted to have a test that can distinguish these cases to make sure my understanding of the `sign_hint` logic is correct.

Reply:
Turns out my understanding wasn't correct, `simplify`, in general, doesn't have to handle width and signedness conversion like this. I just happened to run into a different bug in const folding for case expressions while testing this (working on a fix for that right now) and handling the conversion in simplify worked around that. This should in fact be `newNode = mkconst_int(result, true);` and I'll update this PR when the other bug is fixed.